### PR TITLE
PP-6363 - Send HTTP User agent header to ePDQ 3DS2

### PR DESCRIPTION
**Pull request by nimalank7**
Description:
- This sends the HTTP User agent header to ePDQ 3DS2 for authorisation or else provides a default value

### CODE REVIEW

File: src/test/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2OrderTest.java [hunk 1]
```diff
@@ -203,13 +203,29 @@ public void should_include_browserTimezoneOffSetMins(String timeZoneOffsetMins)
         List<NameValuePair> result = epdqPayloadDefinitionFor3ds2NewOrder.extract(epdqTemplateData);
         assertThat(result, is(aParameterBuilder().withBrowserTimezoneOffsetMins("0").build()));
     }
+
+    @Test
+    public void should_include_provided_browserAgentHeader() {
+        authCardDetails.setUserAgentHeader("Opera/9.8");
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2OrderTest.java:219]: Niche!

File: src/test/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2OrderTest.java [hunk 2]
```diff
@@ -254,7 +275,6 @@ public ParameterBuilder withBrowserTimezoneOffsetMins(String browserTimezoneOffs
                     .add(EXCEPTIONURL_KEY, expectedFrontend3dsIncomingUrl + "?status=error")
                     .add(FLAG3D_KEY, "Y")
                     .add(HTTPACCEPT_URL, ACCEPT_HEADER)
-                    .add(HTTPUSER_AGENT_URL, USER_AGENT_HEADER)
```
Comment (alexbishop1) [inline src/test/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2OrderTest.java:280]: I think this wants to stay. We should probably make sure we do the same normalisation for this one (the defaulting to `Mozilla/5.0` if we don’t get a User-Agent header) as we do for the the new parameter.

Also could you rename `HTTPUSER_AGENT_URL` to `HTTP_USER_AGENT` or something? There’s no need for `URL` to be in there and it confuses me!

**Review by alexbishop1 — CHANGES_REQUESTED** [review]
Looks good — just a couple of things.

File: src/test/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2OrderTest.java (hunk 2 above)
Comment (nimalank7) [reply src/test/java/uk/gov/pay/connector/gateway/epdq/payload/EpdqPayloadDefinitionForNew3ds2OrderTest.java:280]: Done

**Review by alexbishop1 — APPROVED** [review]

**Pull request merged**